lib/elixir/test/elixir/kernel/typespec_test.exs

```diff
@@ -562,6 +575,8 @@ defmodule Kernel.TypespecTest do
       (quote do: @type a_map() :: map()),
       (quote do: @type empty_map() :: %{}),
       (quote do: @type my_map() :: %{hello: :world}),
+      (quote do: @type my_req_map() :: %{required(0) => :atom}),
```

Comment:
It'd be better to use some other atom (`:foo`?) to not confuse it with `atom` type (especially in AST representation `{:atom, _, :atom}`). :bowtie: